src/serializers/type_serializers/datetime_etc.rs

```diff
@@ -37,93 +59,215 @@ fn downcast_date_reject_datetime<'a, 'py>(py_date: &'a Bound<'py, PyAny>) -> PyR
     Err(PydanticSerializationUnexpectedValue::new_from_msg(None).to_py_err())
 }
 
-macro_rules! build_serializer {
```

Comment:
One thing is that this meant this marco wasn't useful anymore. But now all of these serializers follow a very similar pattern. Im not super clued in with marcos so maybe these can all be rewritten to use a marco, but leaving as is for now.

Reply:
I would prefer use a generic to a macro tbh, can help think of ways to factor that out if needed?

Reply:
@davidhewitt done 😄